### PR TITLE
Bump aluminum version

### PR DESCRIPTION
**Pull request by benson31**
Compiled fine against current `llnl/aluminum:master` with all backends. Enabled all bells and whistles in Hydrogen and things were fine.

### CODE REVIEW

**Comment by benson31** [comment]
FWIW, I removed `QUIET` from the `find_package(Aluminum)` calls because CMake outputs a better message than we do. The downside is that one might get false negatives when finding aluminum that (historically) have made some people think that finding aluminum is their build problem when it really isn't. This is just an issue of actually reading all of the CMake output...

**Review by ndryden — APPROVED** [review]
Looks good (once #83 is merged, of course).

**Review by timmoon10 — APPROVED** [review]

**Comment by timmoon10** [comment]
We should make sure to bump versions in the LBANN superbuild and Spack. I'm running into some build errors with `build_lbann_lc.sh` on Pascal.

<details> <summary> HYDROGEN-configure-err.log </summary>

```
CMake Warning at CMakeLists.txt:402 (find_package):
  Could not find a configuration file for package "Aluminum" that is
  compatible with requested version "0.3.0".

  The following configuration files were considered but not accepted:

    /usr/WS1/moon13/src/lbann/build/gnu.Release.pascal.llnl.gov/install/lib64/cmake/aluminum/AluminumConfig.cmake, version: 0.2.1



CMake Error at CMakeLists.txt:427 (message):
  Aluminum support requested but not found.  Please set Aluminum_DIR to point
  to the installation prefix for Aluminum.
```

</details>

**Comment by davidHysom** [comment]
Hm ... this morning I cloned my branch this morning, then pulled in the latest from develop, the built using the script, and had no issues.
________________________________

**Comment by ndryden** [comment]
LBANN pulls 0.2.1 of Aluminum, which is the cause of this. Aluminum 0.3.0 isn't released yet, but will have breaking changes.

**Comment by benson31** [comment]
*sigh*. Ok. I'll peg a new version of Hydrogen that has 0.2.0 as the requirement. I'll set that up to work as the default tag in the superbuild (and by extension, the `build_lbann_lc.sh` script).

In general, `hydrogen` branch should be kept up-to-date with `master` in Aluminum. Also, going forward, I think we should wait until there are about to be breaking changes (in the sense of "there exists an approved PR that will break things when merged") before tagging incompatible versions -- the current `master` branch in Aluminum works just fine with both `hydrogen` of Hydrogen and `develop` of LBANN.

**Pull request merged**
